R.swift/values.swift

```diff
@@ -42,6 +42,75 @@ let ReuseIdentifier = Struct(
   functions: [],
   structs: [])
 
+let StoryboardSegueIdentifierType = Type(name: "StoryboardSegue", genericArgs: ["Segue: UIStoryboardSegue", "Source: UIViewController", "Destination: UIViewController"])
+
+let StoryboardSegueIdentifier = Struct(
+  type: StoryboardSegueIdentifierType,
+  implements: [Type(name: "CustomStringConvertible")],
+  lets: [
+    Let(
+      name: "identifier",
+      type: Type._String
+    )
+  ],
+  vars: [
+    Var(
+      isStatic: false,
+      name: "description",
+      type: Type._String,
+      getter: "return identifier"
+    )
+  ],
+  functions: [
+    Function(
+      isStatic: false,
+      name: "typeSegue",
+      generics: nil,
+      parameters: [
+        Function.Parameter(name: "segue", type: Type(name: "UIStoryboardSegue"))
+      ],
+      returnType: Type(name: "TypedStoryboardSegue", genericArgs: ["Segue","Source","Destination"], optional: true),
+      body: "return TypedStoryboardSegue(segue: segue)"
+    ),
+  ],
+  structs: [])
+
+let TypedStoryboardSegue = Struct(
+  type: Type(name: "TypedStoryboardSegue", genericArgs: StoryboardSegueIdentifierType.genericArgs),
+  implements: [Type(name: "CustomStringConvertible")],
+  lets: [
+    Let(name: "segue", type: Type(name: "Segue")),
+    Let(name: "identifier", type: Type._String.asOptional()),
+    Let(name: "sourceViewController", type: Type(name: "Source")),
+    Let(name: "destinationViewController", type: Type(name: "Destination")),
+  ],
+  vars: [
+    Var(
+      isStatic: false,
+      name: "description",
+      type: Type._String,
+      getter: "return identifier ?? \"\""
+    )
+  ],
+  functions: [
+    Initializer(
+      type: .Designated,
+      isFailable: true,
+      parameters: [
+        Function.Parameter(name: "segue", type: Type._UIStoryboardSegue)
+      ],
+      body: [
+        "guard let segue = segue as? Segue, sourceViewController = segue.sourceViewController as? Source, destinationViewController = segue.destinationViewController as? Destination else { return nil }",
```

Comment:
Good point. Can't do the check here, but will at it in the `typeSegue()` method.